[GLib] Add sysprof profiling documentation

Scony opened the pull request:
#### dcb5f9b201fc42a1c40b5e4a389cc1a1a8f8767f
<pre>
[GLib] Add sysprof profiling documentation
<a href="https://bugs.webkit.org/show_bug.cgi?id=283758">https://bugs.webkit.org/show_bug.cgi?id=283758</a>

Reviewed by Adrian Perez de Castro.

This change introduces a sysprof profiling documentation for GTK and WPE ports.

* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/glib/profiling.md.in: Added.
* Source/WebKit/gtk/gtk3-webkitgtk.toml.in:
* Source/WebKit/gtk/gtk4-webkitgtk.toml.in:
* Source/WebKit/wpe/wpewebkit.toml.in:

Canonical link: <a href="https://commits.webkit.org/287145@main">https://commits.webkit.org/287145@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/36881aebddaf0dd7bc008d3946883d01fb9c081a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/78513 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/57558 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/31895 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/83174 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/29778 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/66709 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/5839 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/61494 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/19411 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/81580 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/51508 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/69585 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/41806 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/48854 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/28115 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/69953 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/25678 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/84540 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/5878 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/4017 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/69721 "Found 3 new test failures: imported/w3c/web-platform-tests/css/css-animations/translation-animation-subpixel-offset.html imported/w3c/web-platform-tests/css/css-pseudo/marker-animate-002.html imported/w3c/web-platform-tests/css/css-transforms/backface-visibility-hidden-animated-001.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/6039 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/67485 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/68975 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17189 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/12995 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/11413 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/5825 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/10269 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/5813 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/9247 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/7600 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WebKit/glib/profiling.md.in:
```diff
@@ -0,0 +1,106 @@
+Title: Profiling
+Slug: profiling
+
+# Profiling with sysprof
```
sysprof → Sysprof.

```suggestion
# Profiling with Sysprof
```

aperezdc commented on Source/WebKit/glib/profiling.md.in:
```diff
@@ -0,0 +1,106 @@
+
+## Prerequisites
+
+To work with sysprof one needs to install it first:
```
Maybe I would make this instance of the word “Sysprof” a linkt to the website:

```suggestion
To work with [Sysprof](https://gitlab.gnome.org/GNOME/sysprof/) one needs to install it first:
```

aperezdc commented on Source/WebKit/glib/profiling.md.in:
````diff
@@ -0,0 +1,106 @@
+```
+sysprof-cli -f
+```
+- or wrap a command with `sysprof-cli` to get a capture funneled to particular process tree created by the given command:
````
Maybe “focused on the execution of a particular process tree” instead of “funneled”?

```suggestion
- or wrap a command with `sysprof-cli` to get a capture focused on the execution of a particular process tree created by the given command:
```

aperezdc commented on Source/WebKit/glib/profiling.md.in:
````diff
@@ -0,0 +1,106 @@
+```
+- or wrap a command with `sysprof-cli` to get a capture funneled to particular process tree created by the given command:
+```
+sysprof-cli -f -- sleep 3
````
Let's put the MiniBrowser here, I think that would be clearer... it took me way too long thinking that `sleep 3` was there to record existing processes during 3 seconds until I realized that `sleep 3` was only an example command:

```suggestion
sysprof-cli -f -- MiniBrowser http://webkit.org
```

aperezdc commented on Source/WebKit/glib/profiling.md.in:
```diff
@@ -0,0 +1,106 @@
+
+### Profiling WebKit in SDK
+
+When profiling WebKit in [SDK](https://github.com/Igalia/webkit-container-sdk), one must use the `Tools/Scripts/run-minibrowser` script as it contains plumbing that makes `sysprof-cli` working correctly:
```
Missing “the”:

```suggestion
### Profiling WebKit in the SDK

When profiling WebKit in the [SDK](https://github.com/Igalia/webkit-container-sdk), one must use the `Tools/Scripts/run-minibrowser` script as it contains plumbing that makes `sysprof-cli` work correctly:
```

aperezdc commented on Source/WebKit/glib/profiling.md.in:
```diff
@@ -0,0 +1,106 @@
+
+Once the sysprof capture file (`.syscap`) is produced using `sysprof-cli`, it's possible:
+
+- to visualize the captured data using sysprof UI
```
```suggestion
- to visualize the captured data using the Sysprof GUI.
```

aperezdc commented on Source/WebKit/glib/profiling.md.in:
````diff
@@ -0,0 +1,106 @@
+(...)
+```
+
+it means that d-bus calls to system bus cannot be authorized. To make sure that's the case, one can run the command:
````
```suggestion
This means that D-Bus calls to the system bus cannot be authorized. To make sure that's the case, one can run the command:
```

aperezdc commented on Source/WebKit/glib/profiling.md.in:
````diff
@@ -0,0 +1,106 @@
+Error org.freedesktop.DBus.Error.AccessDenied: Not authorized to make request
+```
+
+If the above is confirmed, it means that the host system has no active polkit agent that would normally be used to ask user for password (display a GUI popup). In such case one needs to install some polkit agent and run it.
````
```suggestion
This means the host system does not have PolKit agent running that would normally be used to ask the user for their password. In such case one needs to install and run a PolKit agent.
```

aperezdc reviewed (changes requested):
@Scony Thanks a lot for sending this, it's great to see more documentation being added! I left a few comments, mainly small typos and to make some of the phrases a bit easier to parse.

Scony committed:
[GLib] Add sysprof profiling documentation
https://bugs.webkit.org/show_bug.cgi?id=283758

Reviewed by Adrian Perez de Castro.

This change introduces a sysprof profiling documentation for GTK and WPE ports.

* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/glib/profiling.md.in: Added.
* Source/WebKit/gtk/gtk3-webkitgtk.toml.in:
* Source/WebKit/gtk/gtk4-webkitgtk.toml.in:
* Source/WebKit/wpe/wpewebkit.toml.in:

Canonical link: https://commits.webkit.org/287145@main
